test_framework/Jenkinsfile

```diff
@@ -44,9 +44,11 @@ node {
         }
 
         try {
-            stage ('terraform') {
-                sh  " docker exec ${JOB_BASE_NAME}-${BUILD_NUMBER} ${TF_VAR_tf_workspace}/scripts/terraform-setup.sh"
-            }
+            timeout(15) {
```

Comment:
This step will abort terraform if the stage exceeded running over 15 min

Avg running time: ~ 10 min